linode/manager
# refactor: [M3-8981] - Use new preferences selector pattern in more places

## Description 📝

- Uses the new preferences selector pattern introduced in https://github.com/linode/manager/pull/11386 thought Cloud Manager 

## Preview 📷

|  Before  |  After |
|---|---|
| <video src="https://github.com/user-attachments/assets/db6b5d56-381d-4813-9190-d72d7608f2be" />  |  <video src="https://github.com/user-attachments/assets/63ca6b43-9f05-43eb-89aa-25bf2433de7c" />  |
|  All components re-render when a single preference changes  |  With some refactoring and using the new selector pattern, we can localize re-renders 🎉   |

## How to test 🧪

- Verify there are no regressions relating to the touched components
- Use a tool like https://github.com/aidenybai/react-scan to verify re-renders don't cause as many things to re-render when preferences are changed

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
